File: samples/CSharp/Northwind/Region.cs
using System;
using System.Collections.ObjectModel;
using DbExtensions;

namespace Samples.CSharp.Northwind {

   [Table(Name = "Region")]
   public class Region {

      [Column(IsPrimaryKey = true)]
      public int RegionID { get; set; }

      [Column]
      public string RegionDescription { get; set; }

      [Association(OtherKey = nameof(Territory.RegionID))]
      public Collection<Territory> Territories { get; } = new Collection<Territory>();
   }
}

File: samples/VisualBasic/Northwind/Region.vb
Imports System
Imports System.Collections.ObjectModel
Imports DbExtensions

Namespace Northwind

   <Table(Name:="Region")>
   Public Class [Region]

      <Column>
      Property RegionDescription As String

      <Column(IsPrimaryKey:=True)>
      Property RegionID As Integer

      <Association(OtherKey:=NameOf(Territory.RegionID))>
      ReadOnly Property Territories As New Collection(Of Territory)

   End Class

End Namespace
